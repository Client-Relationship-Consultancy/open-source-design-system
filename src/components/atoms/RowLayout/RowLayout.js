import React from "react"
import styled from "styled-components"
import PropTypes from "prop-types"

const StyledRowLayout = styled.div`
  display: flex;
  justify-content: ${props => (props.justifyContent ? props.justifyContent : "flex-start")};
  align-items: ${props => (props.alignItems ? props.alignItems : "flex-start")};
  flex-wrap: ${props => (props.flexWrap ? props.flexWrap : "wrap")};
  > * + * {
    margin-left: 1rem;
  }
`
StyledRowLayout.displayName = "StyledRowLayout"

const RowLayout = ({ id, className, children, justifyContent, alignItems, flexWrap }) => (
  <StyledRowLayout
    id={id}
    className={className}
    justifyContent={justifyContent}
    alignItems={alignItems}
    flexWrap={flexWrap}
  >
    {children}
  </StyledRowLayout>
)
RowLayout.displayName = "RowLayout"

RowLayout.propTypes = {
  id: PropTypes.string,
  className: PropTypes.string,
  justifyContent: PropTypes.string,
  alignItems: PropTypes.string,
  flexWrap: PropTypes.string,
  children: PropTypes.node,
}

export default RowLayout
