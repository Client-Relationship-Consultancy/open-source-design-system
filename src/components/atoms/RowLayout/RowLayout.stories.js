import React from "react"

import { storiesOf } from "@storybook/react"

import { RowLayout, Box } from "../../.."

const stories = storiesOf("Layout", module)

stories.add("Row Layout", () => (
  <RowLayout>
    <Box title="Hello">Hello!!!</Box>
    <Box title="World">Another box, may be wider?</Box>
  </RowLayout>
))
