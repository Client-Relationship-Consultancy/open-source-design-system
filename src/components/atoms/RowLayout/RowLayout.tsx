import React from "react";
import styled from "styled-components";

interface IStyledRowLayout {
  justifyContent?: string;
  alignItems?: string;
  flexWrap?: string;
}

const StyledRowLayout = styled.div<IStyledRowLayout>`
  display: flex;
  justify-content: ${props => (props.justifyContent ? props.justifyContent : "flex-start")};
  align-items: ${props => (props.alignItems ? props.alignItems : "flex-start")};
  flex-wrap: ${props => (props.flexWrap ? props.flexWrap : "wrap")};
  > * + * {
    margin-left: 1rem;
  }
`;
StyledRowLayout.displayName = "StyledRowLayout";

interface IRowLayout extends IStyledRowLayout {
  id?: string;
  className?: string;
  children?: React.ReactNode;
}

const RowLayout: React.SFC<IRowLayout> = ({
  id,
  className,
  children,
  justifyContent,
  alignItems,
  flexWrap,
}) => (
  <StyledRowLayout
    id={id}
    className={className}
    justifyContent={justifyContent}
    alignItems={alignItems}
    flexWrap={flexWrap}
  >
    {children}
  </StyledRowLayout>
);
RowLayout.displayName = "RowLayout";

export default RowLayout;
