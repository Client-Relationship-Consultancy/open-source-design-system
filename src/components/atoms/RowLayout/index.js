import RowLayout from "./RowLayout"

export default RowLayout
