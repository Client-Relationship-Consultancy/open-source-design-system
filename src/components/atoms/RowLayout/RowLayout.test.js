import React from "react"
import { shallow, mount } from "enzyme"
import RowLayout from "./RowLayout"

describe("RowLayout Component Test", () => {
  it("Match last snapshot", () => {
    const component = shallow(<RowLayout>Test</RowLayout>)
    expect(component).toMatchSnapshot()
  })

  it("prop: id - check default and custom value", () => {
    const defaultValue = shallow(<RowLayout />)
    expect(defaultValue.props().id).toEqual(undefined)
    const customValue = defaultValue.setProps({ id: "1" })
    expect(customValue.props().id).toEqual("1")
  })

  it("prop: className - check default and custom value", () => {
    const defaultValue = shallow(<RowLayout />)
    expect(defaultValue.props().className).toEqual(undefined)
    const customValue = defaultValue.setProps({ className: "test" })
    expect(customValue.props().className).toEqual("test")
  })

  it("prop: justifyContent - check default and custom value", () => {
    const defaultValue = mount(<RowLayout />)
    expect(defaultValue).toHaveStyleRule("justify-content", "flex-start")
    const customValue = defaultValue.setProps({ justifyContent: "center" })
    expect(customValue).toHaveStyleRule("justify-content", "center")
  })

  it("prop: alignItems - check default and custom value", () => {
    const defaultValue = mount(<RowLayout />)
    expect(defaultValue).toHaveStyleRule("align-items", "flex-start")
    const customValue = defaultValue.setProps({ alignItems: "center" })
    expect(customValue).toHaveStyleRule("align-items", "center")
  })

  it("prop: flexWrap - check default and custom value", () => {
    const defaultValue = mount(<RowLayout />)
    expect(defaultValue).toHaveStyleRule("flex-wrap", "wrap")
    const customValue = defaultValue.setProps({ flexWrap: "no-wrap" })
    expect(customValue).toHaveStyleRule("flex-wrap", "no-wrap")
  })
})
