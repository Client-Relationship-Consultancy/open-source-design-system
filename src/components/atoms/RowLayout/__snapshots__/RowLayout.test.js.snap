// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RowLayout Component Test Match last snapshot 1`] = `
<StyledRowLayout>
  Test
</StyledRowLayout>
`;
